zebra/zebra_vty.c

```diff
@@ -942,11 +942,17 @@ static void do_show_route_helper(struct vty *vty, struct zebra_vrf *zvrf,
 					if (!tableid)
 						vty_out(vty, "VRF %s:\n",
 							zvrf_name(zvrf));
-					else
-						vty_out(vty,
-							"VRF %s table %u:\n",
```

Comment:
tables have a dual nature in linux.  It's a table and that table also can be a vrf.  I don't think that it's a distinction that we should be making, though.  If I have table 99 and it also is being used by vrf GREEN.  I don't think when I say display `table 99` that we should say it's vrf GREEN in that case, the operator didn't ask for that data.  I don't think it makes sense in the l3mdev based vrf case that we should be displaying anything about a vrf when you display a specific table.

Reply:
as I say, maybe I'm being naive - but aren't there just two possibilities? either a table is in the default vrf, or it's in a different vrf. why not show that association? we're willing to show the vrf and table in the output of `show vrf`, and that seems ... fine. why not here too?

Reply:
I kinds like the fact that if the table happens to be a vrf, then show me that info too.

Reply:
It is an important piece of info. I am with @mjstapp here, I think we should keep it.

Reply:
it doesn't do that Jafar.  If using l3mdev it always displays `Default`.  My point is that default is nonsensical.  The fact that we are asking about a table means that we are not talking about vrfs at all.  In linux they just happen to have taken the l3mdev device and overlayed it over the table id space.  But table 3939 is not in the default vrf, as that it has no relation to the default vrf at all, hence my removal

Reply:
gosh - we seem to be crossed-up here.
none of us thinks it's correct to print "default vrf" when ... it's not - but it would be nice to print the correct vrf, whether default or not (imo).

Reply:
I disagree, the operator has asked for a table not a vrf in this case.  Don't display anything about a vrf then

Reply:
dropping `default`, and keeping the correct vrf when there is actually one is the right approach IMO.